signer/tests/integration/block_observer.rs

```diff
@@ -643,6 +647,195 @@ async fn block_observer_stores_donation_and_sbtc_utxos() {
+                amount: signers_amount,
+                public_key: signers_public_key1,
+            },
+            fee_rate: 2.0,
+            // This ensures that the new signer UTXO is locked by the new
+            // aggregate key.
+            public_key: signers_public_key2,
+            last_fees: None,
+            magic_bytes: [b'T', b'3'],
+        },
+        accept_threshold: 2,
+        num_signers: 3,
+        sbtc_limits: SbtcLimits::unlimited(),
+        max_deposits_per_bitcoin_tx: 25,
+    };
+
+    // By playing around with the votes above, we set things up so that we
+    // get three transactions.
+    let mut transactions = requests.construct_transactions().unwrap();
+    assert_eq!(transactions.len(), 3);
+
+    // We depend on the fact that `construct_transactions` does not sort
+    // the items. In the first transaction, all inputs are locked by the
+    // second key, while the other ones are locked by the new aggregate
+    // key.
+    //
+    // TODO(1661): Technically the packager tries to sort things right now,
+    // but all requests have the same number of votes against so the
+    // current sorting implementation leaves the elements unchanged (we use
+    // a stable sorting algorithm).
+    testing::set_witness_data(&mut transactions[0], signer.keypair);
+    testing::set_witness_data(&mut transactions[1], new_signer.keypair);
+    testing::set_witness_data(&mut transactions[2], new_signer.keypair);
+
+    rpc.send_raw_transaction(&transactions[0].tx).unwrap();
+    rpc.send_raw_transaction(&transactions[1].tx).unwrap();
+    rpc.send_raw_transaction(&transactions[2].tx).unwrap();
+
+    let ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_first_bitcoin_core_client()
+        .with_mocked_emily_client()
+        .with_mocked_stacks_client()
+        .modify_settings(|settings| settings.signer.bootstrap_aggregate_key = Some(aggregate_key))
+        .build();
+
+    let block_observer = BlockObserver {
+        context: ctx.clone(),
+        bitcoin_blocks: (),
+    };
+
+    // Let's make sure the relevant tables are empty
+    let sql = r#"
+        SELECT COUNT(*)::INTEGER
+        FROM bitcoin_tx_inputs
+        WHERE prevout_type = 'signers_input'
+
+        UNION ALL
+
+        SELECT COUNT(*)::INTEGER
+        FROM bitcoin_tx_inputs
+        WHERE prevout_type = 'deposit'
+
+        UNION ALL
+
+        SELECT COUNT(*)::INTEGER
+        FROM bitcoin_tx_outputs
+        WHERE output_type = 'signers_output'"#;
+
+    let num_rows = sqlx::query_scalar::<_, i32>(sql)
+        .fetch_all(db.pool())
+        .await
+        .unwrap();
+
+    assert_eq!(num_rows[0], 0);
+    assert_eq!(num_rows[1], 0);
+    assert_eq!(num_rows[2], 0);
+    assert_eq!(num_rows.len(), 3);
+
+    // Let's generate a new block with the sweep transactions and feed it
+    // to the block observer
+    let block_hash = faucet.generate_blocks(1).pop().unwrap();
```

Comment:
nit:
```suggestion
    let block_hash = faucet.generate_block();
```

Reply:
Done in https://github.com/stacks-sbtc/sbtc/pull/1669/commits/8425c5e55b9bf8d2137a964d0f5e32d1b3f13681